app/charts/map/map-state.tsx

```diff
@@ -578,6 +581,14 @@ const useMapState = (
   };
 };
 
+export const getMapStateMetadata = (): ChartStateMetadata => {
```

Comment:
I think we could have the same method name (getMapStateMetadata -> getMetadata) across all the chart files so that we can load the module async and just call getMetadata on it. This  way, it is even clearer that a chart file needs to implement the getMetadata method.